app/presenters/outcome_presenter.rb

```diff
@@ -74,4 +73,19 @@ def render_data_partial(partial, variable_name)
     partial_path = ::SmartAnswer::FlowRegistry.instance.load_path.join("data_partials", "_#{partial}")
     ApplicationController.new.render_to_string(file: partial_path.to_s, layout: false, locals: {variable_name.to_sym => data})
   end
+
+  def render_erb_template
+    unless @rendered_erb_template
```

Comment:
I've added another commit to this branch ("Avoid instantiating OutcomePresenter multiple times") that should address your concern. The combination of `SmartAnswerPresenter#current_state` being memoized and by now passing the instance of `OutcomePresenter` to the '_result.html.erb' partial, we can be sure that the `state` object within the `OutcomePresenter` won't change.